src/components/InputTag/InputTag.tsx

```diff
@@ -1,16 +1,44 @@
-import { DetailedHTMLProps, InputHTMLAttributes } from "react"
-import style from "./InputTag.module.scss"
+import { DetailedHTMLProps, InputHTMLAttributes } from "react";
+import style from "./InputTag.module.scss";
+import { isNumberValid } from "./validation";
 
 type Props = {
-    label: string,
-    isOn: boolean,
-} & DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>
+    id: string;
+    disabled: boolean;
+    onChange: (state: number) => void;
+} & Omit<
+    DetailedHTMLProps<InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>,
+    "onChange" | "disabled"
+>;
 
-export function InputTag({label, isOn, ...inputProps}: Props) {
+const onChangeInput = (
+    e: React.FormEvent<HTMLInputElement>,
+    onChange: (state: number) => void
+) => {
+    const currentNumber = Number.parseFloat(e.currentTarget.value);
+    if (isNumberValid(e.currentTarget.value, "int64")) {
```

Comment:
Why int64?

Reply:
I changed it to a float64, is it always going to accept a float number? Check type is done in the hook, for that reason I asume it accepts it